### ISSUE TITLE
Unresponsive when launching through NickelMenu on Kobo Libra 2

### ISSUE THREAD

**Issue by gwoob** [Rain92/UltimateMangaReader#73]
When I launch UltumateMangaReader through NickelMenu my Libra 2 goes unresponsive and requires a hard restart to regain responsiveness.

Latest version of everything.

**Comment by Hirngreifer** [comment]
I do have the same Problem :C

**Comment by Beautjess** [comment]
Same here, did everything to fix it but it just freezes on launch.. need to hard reset the Libra2 to get it functional again..

The log says its a unkown device or something.

For the love of everything holly, i hope it gets fixed

**Comment by Hirngreifer** [comment]
Yes but I don't think he works on this project anymore at least not in the near future.

**Comment by tristanlirette** [comment]
Same here. Worth noting that /mnt/onboard/.adds/UltimateMangaReader/logs.txt does not contain any logs after this happens.

**Comment by sjsamphex** [comment]
Same issue on the new Kobo Libra Color

**Comment by kuzniarz** [comment]
The problem sadly still persists on the newest release with Kobo Libra 2! Is someone working on this?

**Comment by mehanix** [comment]
@kuzniarz i'm trying to take a look at it, but am not very experienced with C++ so I have no idea if I'll figure out what's wrong. I'm on Kobo Libra Color so i may need your help testing it on the Libra 2, maybe the issue's different.

**Comment by mehanix** [comment]
Ok after building from source (was a huge pain) with debug symbols enabled and trying to make some minor changes, I'm getting some error logs:
```
[FBInk] Unidentified Kobo device code (390)!
kobofb: Finished initialization.
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
Checking internet connection...
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
ping: sendto: Network is unreachable
Connecting to Wifi...
"INTERFACE: wlan0 WIFI_MODULE: ar6000 PLATFORM: freescale PRODUCT: monza\n"
"[enable-wifi.sh] Closed fd 11 -> /dev/fb0\n"
"[enable-wifi.sh] Closed fd 6 -> /dev/input/event0\n"
"[enable-wifi.sh] Closed fd 8 -> /dev/input/event1\n"
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
"/bin/sh: "
"//: Permission denied"
"\n"
"dhcpcd[23640]: dhcpcd not running\n"
"/bin/sh: "
"//: Permission denied"
"\n"
"dhcpcd[23648]: version 6.6.6 starting\n"
"dhcpcd[23648]: wlan0: using /sys hwaddr: a4:3c:d7:54:28:53\n"
"dhcpcd[23648]: wlan0: executing `/libexec/dhcpcd-run-hooks' PREINIT\n"
"dhcpcd[23648]: wlan0: executing `/libexec/dhcpcd-run-hooks' NOCARRIER\n"
"dhcpcd[23648]: wlan0: waiting for carrier\n"
[FBInk] [refresh_kobo] MXCFB_SEND_UPDATE_V1_NTX: Interrupted system call!
[FBInk] [fbink_refresh] Failed to refresh the screen!
```

This is on Kobo Libra Color, previously to this log.txt was empty so I think i am onto something. I'll try over the next few weeks to figure out what's going on. Wish me luck 😅

**Comment by kuzniarz** [comment]
@mehanix Best of luck! You got this! Let me know if I can be of any help. I'll for sure be able to at least verify the same behavior on my Kobo Libra 2 or test your fix, if you're able to get it working!

**Comment by mehanix** [comment]
@kuzniarz After updating fbink to the latest version in the platform plugin and moving to qt 5.15.2 i got it to boot on my Libra Color without freezing my system! The screen's unresponsive though but it's not frozen (responds to magnetic cover). Progress, certainly, but still a long way to go. I expect it'll take me a few months to get everything in a nice enough shape, if i can figure out the critical issues.

I'll have to note these steps all down so they're reproducible 😅 

![Image](https://github.com/user-attachments/assets/49a9b65c-6bdb-47c2-87cf-390117ad7f34)
